sdks/python/apache_beam/runners/portability/stager.py

```diff
@@ -84,6 +85,8 @@
 WORKFLOW_TARBALL_FILE = 'workflow.tar.gz'
 REQUIREMENTS_FILE = 'requirements.txt'
 EXTRA_PACKAGES_FILE = 'extra_packages.txt'
+# Filename that stores the submission environment dependencies.
+SUBMISSION_ENV_DEPENDENCIES_FILE = 'submission_environment_dependencies.txt'
```

Comment:
Why do we special case submission environment. Should we document/log dependencies of every environment that is used in a pipeline ?

Reply:
There maybe conflation of what we mean by environment.

The intent is to capture the content of environment that creates the translates job graph from Beam Graph to RunnerAPI  proto. In a pure-python pipeline, it is the the envirionment of Python interpreter that calls `pipeline.run()`. In xlang, it is where transform expansion happens. Good point that there might be more than 1 environment involve (if someone had a  XLang transform with two python environments).

On the execution side, we should make `pip freeze` content available to respective runtime environment that will run transform code.